File: frontend-new/src/api/useAutoApi/env.ts
import request from '/@/utils/request';

/**
 * 项目使用接口
 * @method getProjectList 获取项目列表
 * @method getMenuTest 获取后端动态路由菜单(test)
 */
export function useEnvApi() {
  return {
    getList: (data?: object) => {
      return request({
        url: '/env/list',
        method: 'POST',
        data,
      });
    },
    getEnvById: (data?: object) => {
      return request({
        url: '/env/getEnvById',
        method: 'POST',
        data,
      });
    },
    saveOrUpdate(data?: object) {
      return request({
        url: '/env/saveOrUpdate',
        method: 'POST',
        data
      })
    },
    deleted: (data?: object) => {
      return request({
        url: '/env/deleted',
        method: 'POST',
        data,
      });
    },
    getDataSourceByEnvId: (data?: object) => {
      return request({
        url: '/env/getDataSourceByEnvId',
        method: 'POST',
        data,
      });
    },
    bindingDataSource: (data?: object) => {
      return request({
        url: '/env/bindingDataSource',
        method: 'POST',
        data,
      });
    },
    unbindingDataSource: (data?: object) => {
      return request({
        url: '/env/unbindingDataSource',
        method: 'POST',
        data,
      });
    },
  };
}

File: backend_new/autotest/apis/api/env.py
from fastapi import APIRouter

from autotest.corelibs.http import partner_success
from autotest.schemas.api.env import EnvQuery, EnvId, EnvIn, BindingDataSourceIn
from autotest.services.api.env import EnvService, EnvDataSourceService

router = APIRouter()


@router.post('/list', description="环境列表")
async def env_list(params: EnvQuery):
    data = await EnvService.list(params)
    return partner_success(data)


@router.post('/getEnvById', description="获取列表")
async def get_env_by_id(params: EnvQuery):
    data = await EnvService.get_env_by_id(params)
    return partner_success(data)


@router.post('/saveOrUpdate', description="更新保存环境信息")
async def save_or_update_env(params: EnvIn):
    data = await EnvService.save_or_update(params)
    return partner_success(data)


@router.post('/deleted', description="删除环境")
async def delete_env(params: EnvId):
    data = await EnvService.deleted(params)
    return partner_success(data)


@router.post('/getDataSourceByEnvId', description="根据env_id获取数据源")
async def get_data_source_by_env_id(params: EnvId):
    data = await EnvDataSourceService.get_by_env_id(params)
    return partner_success(data)


@router.post('/bindingDataSource', description="绑定关联数据源")
async def binding_data_source(params: BindingDataSourceIn):
    data = await EnvDataSourceService.binding_data_source(params)
    return partner_success(data)


@router.post('/unbindingDataSource', description="解绑关联数据源")
async def unbinding_data_source(params: BindingDataSourceIn):
    data = await EnvDataSourceService.unbinding_data_source(params)
    return partner_success(data)
